Variable Leptonica_LIBRARIES does not know about its dependencies which are handled only
by exported cmake/pc files which are not used by Conan.
Therefore link with exported target from the autogenerated CMake file by the cmake_find_package
that contains information about all dependencies
diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -491,8 +491,8 @@ elseif (SW_BUILD)
     export(TARGETS libtesseract APPEND FILE ${CMAKE_CURRENT_BINARY_DIR}/TesseractTargets.cmake)
 else()
     target_link_libraries       (libtesseract PUBLIC
-        ${Leptonica_LIBRARIES}
-        ${LibArchive_LIBRARIES}
+        Leptonica::Leptonica
+        LibArchive::LibArchive
     )
     export(TARGETS libtesseract FILE ${CMAKE_CURRENT_BINARY_DIR}/TesseractTargets.cmake)
 endif()
